src/components/infinite-data-table.jsx

```diff
@@ -0,0 +1,427 @@
+      </div>
+    );
+  }
+
+  static getCell({
+    column,
+    row,
+    style,
+    rowIndex,
+    showHeader,
+    idKey,
+    onSelect,
+    onHeaderClick,
+    selectedRows,
+  }) {
+    if (showHeader && rowIndex === 0) {
+      return InfiniteDataTableCore.getHeaderCell({
+        column,
+        onHeaderClick,
+        style,
+      });
+    }
+
+    let className = `table-data ${rowIndex % 2 ? 'table-row-odd' : 'table-row-even'}`;
+
+    if (!row) {
+      return InfiniteDataTableCore.getLoadingCell({ className, style });
+    }
+
+    if (column.selectColumn) {
+      return InfiniteDataTableCore.getSelectCell({
+        id: row[idKey],
+        onSelect,
+        selectedRows,
+        className,
+        style,
+      });
+    }
+
+    const id = row[idKey];
+    const checked = id in selectedRows;
+
+    if (checked) {
+      className += ' table-row-selected';
+    }
+
+    if (column.numberColumn) {
+      return InfiniteDataTableCore.getNumberCell({
+        style,
+        column,
+        className,
+        rowIndex,
+        showHeader,
+      });
+    }
+
+    if (!('render' in column)) {
+      return InfiniteDataTableCore.getNoRenderCell({ style, column });
+    }
+
+    return (
+      <div className={`${className} ${checked && ' table-row-selected'}`} style={style}>
+        {column.render(row)}
+      </div>
+    );
+  }
+
+  cellRenderer({
+    columnIndex, rowIndex, key, parent, style,
+  }) {
+    const {
+      showHeader, data, columns, selectedRows, idKey, onSelect, onHeaderClick,
+    } = this.props;
+    const { width } = columns[columnIndex];
+    return (
+      <CellMeasurer
+        cache={this.cache}
+        columnIndex={columnIndex}
+        key={key}
+        parent={parent}
+        rowIndex={rowIndex}
+      >
+        {InfiniteDataTableCore.getCell({
+          column: columns[columnIndex],
+          row: data[rowIndex],
+          style: { ...style, width },
+          rowIndex,
+          showHeader,
+          idKey,
+          onSelect,
+          onHeaderClick,
+          selectedRows,
+        })}
+      </CellMeasurer>
+    );
+  }
+
+  isRowLoaded({ index }) {
+    const { data } = this.props;
+    return !!data[index];
+  }
+
+  render() {
+    const {
+      columns,
+      data,
+      onLoadMoreRows,
+      showHeader,
+      fixedColumnCount,
+      fixedRowCount,
+      totalNumberRows,
+    } = this.props;
+    const rowCount = showHeader + Math.min(totalNumberRows, data.length);
+    return (
+      <InfiniteLoader
+        loadMoreRows={onLoadMoreRows}
+        isRowLoaded={this.isRowLoaded}
+        rowCount={rowCount}
+        threshold={1}
+      >
+        {({ onRowsRendered, registerChild }) => (
+          <AutoSizer>
+            {({ width, height }) => (
+              <div className="data-table-infinite">
+                <MultiGrid
+                  cellRenderer={this.cellRenderer}
+                  columnCount={columns.length}
+                  width={width || 100}
```

Comment:
These default width/height could be set in constants

Reply:
Done.

Reply:
Done.